src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -477,6 +482,23 @@ CameraTrigger::cycle_trampoline(void *arg)
 				cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED;
 
 			}
+
+		} else if (cmd.command == 12345 /* VEHICLE_CMD_SEND_CAMERA_IMAGE_META */) {
+
+			// This command is designed to allow the GCS / ground to re-request
+			// trigger packets that might have dropped over the link.
+			// The GCS can infer this when the trigger sequence skips once. It should
+			// store the triggers on the ground in a hash map using the sequence as
+			// hash. Then simply checking the last received index allows for an
+			// efficient retransmission logic.
+
+			if (static_cast<int>(cmd.param1) == 0 /*_prev_image.seq*/) {
+				// Send prev image again
```

Comment:
We are currently using a similar mechanism to get the tags to the GCS. And I can confirm that it happens every now and then that you lose the mavlink connection for a longer time. Especially when you start to fly further away and BVLOS.

We found a workaround to store the tags using the datamanager during the flight. Then only after the flight we transmit the full set of tags. As the tags are only little data and we don't save them in junks this proofed to work reliably.